### ISSUE TITLE
Solved: Rufus 'killed' my USB Drive

### ISSUE THREAD

**Issue by WI-Hedgehog** [pbatard/rufus#2105]
﻿

Checklist
---------
- [x] I looked at https://github.com/pbatard/rufus/wiki/FAQ to see if my question has already been answered.
- [x] I performed a search in the issue tracker for similar issues using keywords relevant to my problem, such as the error message I got from the log.
- [ ] I clicked the 'Log' button or pressed <kbd>Ctrl</kbd>-<kbd>L</kbd> in Rufus, and copy/pasted the log into the line that says `<FULL LOG>` below.
- [ ] The log I am copying is the FULL log, starting with the line `Rufus version: x.y.z` - I have NOT removed any part of it.

Additionally (if applicable):
- [x] I ran a bad blocks check, by clicking _Show advanced format options_ then _Check device for bad blocks_, and confirmed that my USB is not defective.
- [x] I also tried one or more of the following:
  - [ ] Using a different USB drive.
  - [x] Plugging the USB into a different port.
  - [x] Running Rufus on a different computer.
- [x] If using an image, I clicked on the `(✓)` button to compute the MD5, SHA1 and SHA256 checksums, which are therefore present in the log I copied. I confirmed, by performing an internet search, that these values match the ones from the official image.

Issue description
-----------------
Hi Pete, I have a potential solution (not a complaint). There's a thread on here "Rufus killed my USB Drive" that is closed, and many threads elsewhere where people are having similar issues. I had this issue where it appeared Rufus "killed" my USB flash drive, however it did not and there may be an easy but not apparent solution.

1.) Rufus worked fine putting a SystemRescue 750 MB ISO image 
  on a 32GB Toshiba flash drive (new-ish with low use, reliable), 
  via a USB3 port. 
  It was very fast and the flash drive got fairly warm.
+Using the flash drive worked fine in several systems.

2.) Rufus worked fine re-writing a Rescuezilla 1.1 GB ISO image  
  on the same flash drive, 
  USB3, 
  drive was quite warm.
+Using the flash drive worked fine in several systems.

3.) Later re-writing a Rescuezilla 1.1 GB ISO image via Rufus 
  same drive, 
  USB3, 
  drive was warm enough to be concerned it should sit a few minutes and cool down.
  As previous, the write threw no errors.
-However, the image failed to load in several systems with what looked like a flash memory corruption issue.

4.) Rewriting the same Rescuezilla 1.1 GB ISO image via Rufus 
  same drive, 
  USB3, 
  warm drive,
  no write errors.
-The ISO image on the flash drive created worse corruption results when loading. It looked like flash memory was going bad.

5.) Using Windows 10 quick format and 
  retrying writing the Rescuezilla 1.1 GB ISO image via Rufus 
  on -=USB2=- port 
  warmed the drive up some,
  no reported write errors (which is still consistent)
-created even *worse* corruption results than before.

6.) Clearing the USB via Rufus and 
  retrying writing the 1.1 GB ISO
  on USB2
  warmed drive,
  no reported errors,
-even worse corruption results than before.

7.) Copying files to the drive via Windows File Explorer works fine, -=no corruption=-.

Obviously at this point it looks like Rufus is the problem, 
8.) but using Rufus to do a test of the flash drive on both USB2 and USB3 ports showed the drive was fine.
9.) Rufus could be incorrect, but the files could be written and read by Windows fine if the USB flash drive was used for file storage (not booting an OS).

10.) I used Ventoy and had "different" boot issues. After reading the manuals I used Ventoy to create 4GB of slack space at the end of the thumb drive so partitions 1 & 2 were located under 32 GB. The drive now works. This is similar to people reporting they can reformat a USB thumb drive that failed with Rufus and it then works, but yet the drive no longer works with Rufus. 

What's curious is before "shrinking" the flash drive in Ventory (the drive having the full 32 GB of storage space), a sha256sum on the ISO files copied to the flash drive checked fine on the drive with sha256sum reporting no problems, so how can Rufus be causing a worsening problem with the flash drive? 

Shrinking the drive via Ventory did -not- change the file contents or location, 
shrinking moved the "boot partition" (partition #2) down and put unused space behind it, 
and that caused the flash drive to load properly.

The problem seems to be with how the computer BIOS reads the USB flash drive. 
A documented example is many HP Proliant servers (and many other desktop systems) 
cannot load from thumb drives larger than 32 GB, 
and they seem to have mixed issues with 32 GB USB thumb drives (even using work-arounds). 
Decrease the size to under 32 GB and they work fine. (Even my newer Gigabyte mainboard has issues with the 32 GB flash drive.)

Separately, I do not doubt Rufus writing to a flash drive at high speed and therefore significantly heating the drive can, at times, make it look like Rufus "killed the drive." Some flash drives fail due to excessive heat or one or more of several other usage-related reasons, though that is not directly related to Rufus--it is related to using the drive in general. Flash drives do expire, even when new, similar to how new hard drives occasionally suffer from infant mortality.
-BUT-
It is the case from so many reports and my own experience Rufus has some impact on how USB flash data is structured. However, if the drive is able to hold data outside of Rufus writing to it--and without the data being corrupted--then the drive hasn't been "killed," something else is going on.

Thank you for the software Pete, it works quite well. Sometimes it works better with a seemingly smaller drive depending on the computer's BIOS. I've tested and re-tested this theory since originally posting this theory and while not enough testing has been done for the theory to be conclusive, it does seem plausible.

--Hedge

Log
---
(system hung, could not provide log)
<FULL LOG>
(system hung, could not provide log)

**Comment by pbatard** [comment]
Thanks for the input.

I'm suspecting that part of the issue may be related to using Large FAT32, as, officially (i.e. when using the Microsoft formatting tools), FAT32 can only officially accommodate drives up to 32 GB, and anything larger than that requires using _Large FAT32_ which is not technically something Microsoft supports (NB: Rufus did not invent Large FAT32 &mdash; we merely reused the well known formatting algorithm from RidgeCrop software).

In short, it's not unlikely that some pedantic BIOSes may have trouble with volumes larger than 32 GB using FAT32, as it's not something that officially supported by Microsoft software, and I suppose this comes into play with what you report above.

Hopefully your entry will help others who run into these kind of problems. I will close this issue now.
